clang/lib/Format/Format.cpp

```diff
@@ -1647,7 +1647,7 @@ FormatStyle getLLVMStyle(FormatStyle::LanguageKind Language) {
   LLVMStyle.SeparateDefinitionBlocks = FormatStyle::SDS_Leave;
   LLVMStyle.ShortNamespaceLines = 1;
   LLVMStyle.SkipMacroDefinitionBody = false;
-  LLVMStyle.SortIncludes = {/*Enabled=*/true, /*IgnoreCase=*/false};
+  LLVMStyle.SortIncludes = {/*Enabled=*/true, /*IgnoreCase=*/false, /*IgnoreExtension=*/false};
```

Comment:
if I understand correctly... true is the current order not false, so the default should be true otherwise you'll cause every user of clang-format to have to add the IgnoreExtension setting if they don't want a change, am I correct?

Reply:
@mydeveloperday The default is false, by default, the extension is not ignored.

Reply:
in the documentation you show, neither true/false gives you what clang-format gives us today (with just BasedOnStyle: LLVM

which is ...

```c++
#include "A.h"
#include "A-util.h"
#include "A.inc"
```

So lets assume I don't what your feature... how do I keep my current format? what should it be set to true or false?

```
       true:                      false:
       # include "A.h"             # include "A-util.h"
       # include "A.inc"           # include "A.h"
       # include "A-util.h"        # include "A.inc"
```

Reply:
> "currently will always put analyze.h last after all the analyze-xxx.h headers, but putting analyze.h first instead of last is arguable nicer to read"

How come for me its doesn't do this as you say... 

seems your example might not be the best for the documentation, seems like this is dependent on the basename length being greater than 1 in length. Ok so thats confusing...the documentation should represent what happens.

```c++
#include "ab-beta.h"
#include "ab-data.h"
#include "ab.h"
```
```c++
#include "a.h"
#include "a-beta.h"
#include "a-data.h"
```

Reply:
@mydeveloperday Huh, I don't see how the length of the basename would affect this. I pushed more tests to verify this is the case.

Reply:
```
#include "A.h"
#include "A-util.h"
#include "A.inc"
```

I don't see how clang-format would ever return this today without disabling include sorting? `-` sorts before `.` so `A-util.h` will always be put before `A.h` and `A.inc` today if include sorting is enabled.

Reply:
Is the file maybe `A.cpp`?

Reply:
yes.. thats it.. it was A.cxx actually..

Reply:
Currently it wouldn't change anything until the option is enabled. For me this change is good, I did just not vote, because of @mydeveloperday request for changes.

And as cleared, the the "wrong" sorting was due using `A.cpp` as filename and thus `A.h` becomes the main header and is put up front.

Reply:
Correct, either @mydeveloperday says what has to be done, accepts this patch, or we overrule him.